src/helpers/wallet.helper.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { HathorWallet, walletUtils, Connection, errors } from '@hathor/wallet-lib';
+import config from '../config';
+
+export class WalletStartError extends Error {}
+
+export function getReadonlyWallet({
+  xpub,
+  multisigData = null,
+  preCalculatedAddresses = [],
+}) {
+  const connection = new Connection({
```

Comment:
The connection is the same for both methods. Why don't you create before and have it as a parameter?

Reply:
Done